specification/dataprotection/resource-manager/Microsoft.DataProtection/stable/2023-01-01/dataprotection.json

```diff
@@ -0,0 +1,7082 @@
+      "get": {
+        "tags": [
+          "ResourceGuards"
+        ],
+        "summary": "Returns ResourceGuards collection belonging to a ResourceGroup.",
+        "operationId": "ResourceGuards_GetResourcesInResourceGroup",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v4/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v4/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v4/types.json#/parameters/ResourceGroupNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ResourceGuardResourceList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Get ResourceGuards in ResourceGroup": {
+            "$ref": "./examples/ResourceGuardCRUD/GetResourceGuardsInResourceGroup.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataProtection/resourceGuards/{resourceGuardsName}": {
+      "put": {
+        "tags": [
+          "ResourceGuards"
+        ],
+        "summary": "Creates or updates a ResourceGuard resource belonging to a resource group.",
+        "operationId": "ResourceGuards_Put",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v4/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v4/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v4/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "resourceGuardsName",
+            "in": "path",
+            "description": "The name of ResourceGuard",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "Request body for operation",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ResourceGuardResource"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
PUT request should be able to return 201 as well as 200 if it supports 'create or update'

Reply:
Addressed